app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt

```diff
@@ -0,0 +1,435 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.CheckCircle
+import androidx.compose.material.icons.outlined.DoNotDisturbOn
+import androidx.compose.material.icons.outlined.FilterAlt
+import androidx.compose.material.icons.outlined.GroupAdd
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Tab
+import androidx.compose.material3.TabRow
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+
+@Composable
+fun FriendsScreen(
+    model: FriendsViewModel = hiltViewModel()
+) {
+    val context = LocalContext.current
+    val toastMessage by model.toastMessage
+    val friendObject by model.friendObject.collectAsStateWithLifecycle()
+    val currentUserEmail = model.user.collectAsStateWithLifecycle().value?.email
+    val friends = friendObject?.friends ?: emptyList()
+    val isRequestSent by model.isRequestSent.collectAsState(false)
+    val receivedRequests = friendObject?.receivedRequests ?: emptyList()
+    val sentRequests = friendObject?.sentRequests ?: emptyList()
+    val friendEmail = remember { mutableStateOf("") }
+    val showAddFriend = remember { mutableStateOf(false) }
+    val showRequests = remember { mutableStateOf(false) }
+    val selectedTab = remember { mutableIntStateOf(0) }
+    val scrollState = rememberScrollState()
+
+    LaunchedEffect(toastMessage) {
+        toastMessage?.let {
+            Toast
+                .makeText(
+                    context,
+                    it,
+                    Toast.LENGTH_SHORT
+                )
+                .show()
+            model.toastMessage.value = null
+        }
+    }
+
+    LaunchedEffect(key1 = Unit) {
```

Comment:
Could drop the `key1 =` part, it is the first parameter, will be cleaner to read.